diff --git a/python/CMakeLists.txt b/python/CMakeLists.txt
index 4016f7c..1e45797 100644
--- a/python/CMakeLists.txt
+++ b/python/CMakeLists.txt
@@ -26,8 +26,12 @@ SET(_swig_files
   medsubdomain_module.i
 )
 
+IF(APPLE)
+  SET(PYTHON_LIBRARIES "-undefined dynamic_lookup")
+ENDIF(APPLE)
+
 SET(_link_libs
-  med
+  medC
   ${PYTHON_LIBRARIES}
   )
 
diff --git a/python/medenum_module.i b/python/medenum_module.i
index 91fc0d8..920c9eb 100644
--- a/python/medenum_module.i
+++ b/python/medenum_module.i
@@ -6,6 +6,7 @@
 
 %{
 #include "med.h"
+#include <utility>
 %}
 
 %include "H5public_extract.h"
diff --git a/python/medenumtest_module.i b/python/medenumtest_module.i
index 7bfc32b..efda37d 100644
--- a/python/medenumtest_module.i
+++ b/python/medenumtest_module.i
@@ -4,6 +4,7 @@
 
 %{
 #include "med.h"
+#include <utility>
 %}
 
 %include "H5public_extract.h"
